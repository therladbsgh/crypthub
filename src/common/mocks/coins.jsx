export const coins = [
    {
        name: 'Bitcoin',
        symbol: 'BTC',
        currPrice: 10000
    },
    {
        name: 'Ethereum',
        symbol: 'ETH',
        currPrice: 5000
    }
]
