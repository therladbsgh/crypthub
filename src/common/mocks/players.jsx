import date from 'date-and-time';

export const player1 = {
    name: 'player1',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 250
        },
        {
            name: 'Ethereum',
            amount: 1000.28373623
        },
        {
            name: 'USD',
            amount: 250
        }
    ],
    transactions: {
        history: [
            {
                id: '1',
                type: 'market',
                side: 'buy',
                size: 25,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 1 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: true,
                filledDate: date.parse('Apr 3 2018', 'MMM D YYYY')
            },
            {
                id: '2',
                type: 'limit',
                side: 'buy',
                size: 35.000006,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('Apr 5 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ],
        current: [
            {
                id: '3',
                type: 'short',
                side: 'sell',
                size: 80,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: false,
                filledDate: null
            },
            {
                id: '4',
                type: 'limit',
                side: 'buy',
                size: 2,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 7 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('May 2 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ]
    },
    netWorth: 200,
    numTrades: 80,
    netReturn: -20,
    todayReturn: 5,
    currRank: 2,
    buyingPower: 150,
    shortReserve: 50
};

export const player2 = {
    name: 'player2',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 300
        }
    ],
    transactions: {
        history: [
            {
                id: '5',
                side: 'buy',
                size: 1,
                price: 10,
                symbol: 'ETH'
            },
            {
                id: '6',
                side: 'sell',
                size: 3500,
                price: 202220,
                symbol: 'ETH'
            }
        ],
        current: [
            {
                id: '7',
                side: 'sell',
                size: 23230,
                price: 232,
                symbol: 'BTC'
            },
            {
                id: '8',
                side: 'buy',
                size: 6000,
                price: 50,
                symbol: 'ETH'
            }
        ]
    },
    netWorth: 300,
    numTrades: 2,
    netReturn: 120
};
