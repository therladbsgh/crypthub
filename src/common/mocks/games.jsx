import date from 'date-and-time';
import * as PlayerMocks from './players';

export const game1 = {
    id: '1',
    name: 'game1',
    description: 'desc1',
    host: 'user1',
    created: date.parse('Apr 1 2018', 'MMM D YYYY'),
    start: date.parse('Apr 2 2018', 'MMM D YYYY'),
    end: date.parse('May 7 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game2 = {
    id: '2',
    name: 'game2',
    description: 'desc2',
    host: 'user2',
    created: date.parse('Apr 2 2018', 'MMM D YYYY'),
    start: date.parse('Apr 3 2018', 'MMM D YYYY'),
    end: date.parse('May 1 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game3 = {
    id: '3',
    name: 'game3',
    description: 'desc3',
    host: 'user3',
    created: date.parse('Apr 4 2018', 'MMM D YYYY'),
    start: date.parse('Apr 8 2018', 'MMM D YYYY'),
    end: date.parse('Apr 28 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};
