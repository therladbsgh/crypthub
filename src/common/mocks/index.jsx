import * as GameMocks from './games';
import * as PlayerMocks from './players';

export { GameMocks, PlayerMocks };
