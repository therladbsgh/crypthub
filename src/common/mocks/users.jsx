import * as GameMocks from './games';

export const user1 = {
    id: 'id1',
    name: 'user1',
    ELO: 3000,
    games: [
        GameMocks.game1,
        GameMocks.game2
    ]
};

export const user2 = {
    id: 'id2',
    name: 'user2',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};

export const user3 = {
    id: 'id3',
    name: 'user3',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};
