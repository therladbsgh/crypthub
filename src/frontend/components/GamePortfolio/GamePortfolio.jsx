import * as _ from 'lodash';
import React, { Component } from 'react';
import { Header, Button } from 'semantic-ui-react';
import { Portfolio, Transactions } from 'components';

export default class GamePortfolio extends Component {
	constructor(props) {
		super(props);

		this.state = {
			history: false
		};

		this.setHistory = this.setHistory.bind(this);
	}

	setHistory(bool) {
		this.setState({ history: bool });
	}
	
	render() {
		const { player } = this.props;
		const { history } = this.state;
		const transactions = player.transactions;

		return (
			<div>
				<Header as='h2'>Your Portfolio</Header>
				<Portfolio portfolio={player.portfolio} />
				<Header as='h2'>Transactions</Header>
				<Button.Group>
					<Button toggle active={!history} onClick={() => this.setHistory(false)}>Current Orders</Button>
					<Button toggle active={history} onClick={() => this.setHistory(true)}>History</Button>					
				</Button.Group>
				<Transactions transactions={this.state.history ? transactions.history : transactions.current} />
			</div>
		);
	}
}
