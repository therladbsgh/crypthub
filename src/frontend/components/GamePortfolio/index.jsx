import GamePortfolio from './GamePortfolio';

export default GamePortfolio;
