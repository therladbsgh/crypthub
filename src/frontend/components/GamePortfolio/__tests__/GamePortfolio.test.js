import React from 'react';
import GamePortfolio from '../GamePortfolio';
import { GameMocks } from 'mocks';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  	const tree = renderer.create(<GamePortfolio player={GameMocks.game1.players[0]} />).toJSON();
  	expect(tree).toMatchSnapshot();
});