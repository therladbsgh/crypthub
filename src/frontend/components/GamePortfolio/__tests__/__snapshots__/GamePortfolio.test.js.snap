// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2
    className="ui header"
  >
    Your Portfolio
  </h2>
  <table
    className="ui celled table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Symbol
        </th>
        <th
          className=""
        >
          % of Holdings
        </th>
        <th
          className=""
        >
          Amount
        </th>
        <th
          className=""
        >
          Current Price (Today's Change)
        </th>
        <th
          className=""
        >
          Current Value
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className="positive"
      >
        <td
          className=""
        >
          BTC
        </td>
        <td
          className=""
        >
          29.41%
        </td>
        <td
          className=""
        >
          250.00000000
        </td>
        <td
          className=""
        >
          $10,000.00 (+3.25%)
        </td>
        <td
          className=""
        >
          $2,500,000.00
        </td>
      </tr>
      <tr
        className="error"
      >
        <td
          className=""
        >
          ETH
        </td>
        <td
          className=""
        >
          58.83%
        </td>
        <td
          className=""
        >
          1,000.28373623
        </td>
        <td
          className=""
        >
          $5,000.00 (-20.01%)
        </td>
        <td
          className=""
        >
          $5,001,418.68
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className=""
        >
          USD
        </td>
        <td
          className=""
        >
          11.76%
        </td>
        <td
          className=""
        >
          $1,000,000.00
        </td>
        <td
          className=""
        >
          -
        </td>
        <td
          className=""
        >
          $1,000,000.00
        </td>
      </tr>
    </tbody>
    <tfoot
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          <strong>
            TOTAL
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            100.00%
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            -
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            -
          </strong>
        </th>
        <th
          className=""
        >
          <strong>
            $8,501,418.68
          </strong>
        </th>
      </tr>
    </tfoot>
  </table>
  <h2
    className="ui header"
  >
    Transactions
  </h2>
  <div
    className="ui buttons"
  >
    <button
      className="ui active toggle button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Current Orders
    </button>
    <button
      className="ui toggle button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      History
    </button>
  </div>
  <div>
    <br />
    You have no current trade orders.
  </div>
</div>
`;
